### ISSUE TITLE
File size limitation?

### ISSUE THREAD

**Issue by bdcoelho** [hongyuanjia/eplusr#26]
There appears to be a limitation on the size of idf that you can import using read_idf. I just tried a fairly large file (~13 MB) and it fails as follows:

>  model<-read_idf("Test_20181115_182003.idf")
Start parsing...
Error: `file` must be a string, raw vector or a connection.

I tried changing the file name but no difference.

Regards,
Benson

**Comment by hongyuanjia** [comment]
Please make sure file `"Test_20181115_182003.idf"` is in the same directory as `getwd()`.

**Comment by bdcoelho** [comment]
That was fine. It looks like it might be a version problem. My idf is Version 9.0. I tried to update the idd version as follows:

Copied the V 9.0.1 idd to my working directory and entered the following

path_idd<-"C:\\Users\\SIM-PC\\Desktop\\SurfaceR\\V9-0-1-Energy+.idd"
use_idd(path_idd)

That didn't fix the problem. The V 8.8 IDFs run fine, but none of the V 9.0 ones do.
Is there something else that I need to do to update the version, or does it require a code change?


Regards,
Benson

**Comment by hongyuanjia** [comment]
Unfortunately, I cannot reproduce the problem. V9.0 model gets parsed correctly on my side on Windows. Also note that the example file is about 13MB large.

```r
avail_eplus()
# [1] "8.6.0" "8.7.0" "8.8.0" "8.9.0" "9.0.0"

idf <- read_idf("C:/EnergyPlusV9-0-0/ExampleFiles/CmplxGlz_MeasuredDeflectionAndShading.idf")
Idd v9.0.0 has not been parsed before. Try to locate `Energy+.idd` in EnergyPlus v9.0.0 installation folder `C:/EnergyPlusV9-0-0`.
IDD file found: `C:\EnergyPlusV9-0-0\Energy+.idd`.
Start parsing...

  Parsing IDD (Parsing ) [===>-----------------]  20% in  0s
  Parsing IDD (Parsing ) [=====>---------------]  30% in  1s
  Parsing IDD (Parsing ) [=======>-------------]  40% in  1s
  Parsing IDD (Parsing ) [=========>-----------]  50% in  4s
  Parsing IDD (Parsing ) [============>--------]  60% in  5s
  Parsing IDD (Parsing ) [================>----]  80% in  9s
  Parsing IDD (Parsing ) [===================>-]  95% in  9s
  Parsing IDD (Complete) [=====================] 100% in  9s
```

---
> `path_idd<-"C:\Users\SIM-PC\Desktop\SurfaceR\V9-0-1-Energy+.idd"`
> `use_idd(path_idd)`

You have to escape backslash on Windows:

```r
path_idd<-"C:\\Users\\SIM-PC\\Desktop\\SurfaceR\\V9-0-1-Energy+.idd"
# OR use slash
path_idd<-"C:/Users/SIM-PC/Desktop/SurfaceR/V9-0-1-Energy+.idd"
use_idd(path_idd)
```

By the way, where is EnergyPlus v9.0 installed? Please give me results of both `avail_eplus()` and `avail_idd()`.

**Comment by bdcoelho** [comment]
Okay, I think I've found the problem. 
1. I needed a newer version of eplusr
2. I have the bug fixes release of the latest EPLUS -  9.0.1 and eplusr couldn't find the IDD it was looking for which was V 9.0 (doesn't exist on my computer)

I got it by running

use_idd(9.0, download = TRUE)

All looks good now. 

Thanks for your help.

**Comment by hongyuanjia** [comment]
Great! Close now.

Currently, the highest version of EnergyPlus supported by eplusr is v9.0.0. I will add support for EnergyPlus v9.0.1 in the next release. But I believe this bug fixes release should not contain any major modifications on IDD file.

BTW, you can save that downloaded IDD using `download_idd()` on your computer to avoid to download it again next time.
